### PR TITLE
Smaller map graph follow unfollow buttons

### PR DESCRIPTION
**Pull request by yzmopolis**
Important changes:

- I've changed paddings and dimens for `follow`, `unfollow`, `map`, `graphs` buttons
- I've added containers for buttons
- I used TouchDelegate to keep the touchable area the same as earlier 

Screenshot 2021-02-03 at 19 18 39

AC:

- [x] it matches proto: https://projects.invisionapp.com/share/DCWT9W6BUE7#/screens/414476845
- [ ] touchable area is the same

[TouchDelegate docs](https://developer.android.com/reference/android/view/TouchDelegate.html)

[ticket](https://trello.com/c/3fb7fnUc/1070-cards-map-graph-follow-buttons-should-match-the-proto)

### CODE REVIEW

File: app/src/main/res/values/dimens.xml [hunk 1]
```diff
@@ -46,6 +46,10 @@
 
     <dimen name="card_corner_radius">0dp</dimen>
 
+    <dimen name="height_0">0dp</dimen>
+    <dimen name="padding_2">2dp</dimen>
+    <dimen name="margin_top_5">5dp</dimen>
```
Comment (lesniakania) [inline app/src/main/res/values/dimens.xml:51]: when we started implementing layouts with @mturnau we used similar constants, but at some point, we agreed (I think after reading some article, but I don't remember exactly) that when we have a situation that we just need to use specific dimensions, we are just using it in the XML as it is, without adding it to the dimens file.
I guess it was about readability and not making too many and a bit artificial dimens.

File: app/src/main/java/io/lunarlogic/aircasting/screens/dashboard/SessionViewMvcImpl.kt [hunk 2]
```diff
@@ -91,25 +99,34 @@ abstract class SessionViewMvcImpl<ListenerType>: BaseObservableViewMvc<ListenerT
             onCollapseSessionCardClicked()
             collapseSessionCard()
         }
+
         mFollowButton = findViewById(R.id.follow_button)
+        mFollowTouchableArea = findViewById(R.id.follow_container)
```
Comment (lesniakania) [inline app/src/main/java/io/lunarlogic/aircasting/screens/dashboard/SessionViewMvcImpl.kt:104]: I wanted to comment that we agreed on using "the new magic syntax" as I was referring to kotlin synthetics syntax, but I wanted to google it and look what I just found....:
https://medium.com/better-programming/why-are-kotlin-synthetics-deprecated-and-what-are-the-alternatives-5c2b087dda1c /cc @michalrazny2 ....
I guess we should be moving to the data binding syntax, but let's not worry about this now, maybe just when you create any new views, please use that.

File: app/src/main/java/io/lunarlogic/aircasting/screens/dashboard/SessionViewMvcImpl.kt [hunk 3]
```diff
@@ -44,6 +47,11 @@ abstract class SessionViewMvcImpl<ListenerType>: BaseObservableViewMvc<ListenerT
     private var mGraphButton: Button
     private var mLoader: ImageView?
 
+    private var mFollowTouchableArea: LinearLayout
+    private var mUnfollowTouchableArea: LinearLayout
+    private var mMapTouchableArea: LinearLayout
+    private var mGraphTouchableArea: LinearLayout
```
Comment (lesniakania) [inline app/src/main/java/io/lunarlogic/aircasting/screens/dashboard/SessionViewMvcImpl.kt:53]: I'm a bit concerned with these linear layouts - too many layouts (can cause performance problems)[https://stackoverflow.com/a/15405309] and we have lots of sessions on the fixed and dormant tabs to display.
I'd try removing them and in the `expandViewHitArea` use something like this:
```kotlin
childRect.top -= TOUCHABLE_PADDING;
childRect.left -= TOUCHABLE_PADDING;
childRect.right += TOUCHABLE_PADDING;
childRect.bottom += TOUCHABLE_PADDING;
```

where `TOUCHABLE_PADDING` would be some constant defined in this class.
you can check more examples [here](https://www.programcreek.com/java-api-examples/?api=android.view.TouchDelegate).

**Review by lesniakania — CHANGES_REQUESTED** [review]

File: app/src/main/java/io/lunarlogic/aircasting/screens/dashboard/SessionViewMvcImpl.kt (hunk 3 above)
Comment (yzmopolis) [reply app/src/main/java/io/lunarlogic/aircasting/screens/dashboard/SessionViewMvcImpl.kt]: Thanks for this comment, I'll remember it :)

File: app/src/main/res/values/dimens.xml (hunk 1 above)
Comment (yzmopolis) [reply app/src/main/res/values/dimens.xml]: @lesniakania of course, I didn't know, I'll follow your approach 🐾

**Review by lesniakania — APPROVED** [review]

**Pull request merged**